pkgs/build-support/docker/default.nix

```diff
@@ -999,26 +1008,30 @@ rec {
           '';
         };
 
-        closureRoots = lib.optionals includeStorePaths /* normally true */ (
-          [ baseJson customisationLayer ]
-        );
-        overallClosure = writeText "closure" (lib.concatStringsSep " " closureRoots);
-
-        # These derivations are only created as implementation details of docker-tools,
-        # so they'll be excluded from the created images.
-        unnecessaryDrvs = [ baseJson overallClosure customisationLayer ];
+        layersJsonFile = buildPackages.dockerMakeLayers {
```

Comment:
for people who would like to experiment with different ways of grouping paths into layers, this could now be  override (at pkgs level) in order to inject arbitrary code for generating layers 

probably not the most friendly interface but at least it's an improvement over status quo of `streamLayeredImage` being [tightly coupled to referencesByPopularity  approach](https://github.com/NixOS/nixpkgs/blob/172cb3ef53e126bf2a81b00667e5c0c351dd0476/pkgs/build-support/docker/default.nix#L1068-L1084)

I'd be happy to iterate on this in order to make the injection of layering logic more user friendly.